ArchiSteamFarm/Steam/Integration/SteamChatMessage.cs

```diff
@@ -0,0 +1,193 @@
+				if (line.Length == 0) {
+					if (messagePart.Length == 0) {
+						if (prefixBytes > 0) {
+							messagePartBytes += prefixBytes;
+							messagePart.Append(steamMessagePrefix);
+						}
+					} else {
+						messagePartBytes += NewlineWeight;
+						messagePart.AppendLine();
+					}
+
+					// Check if we reached the limit for one message
+					if (messagePartBytes + NewlineWeight + ReservedEscapeMessageBytes > maxMessageBytes) {
+						yield return messagePart.ToString();
+
+						messagePartBytes = 0;
+						messagePart.Clear();
+					}
+
+					// Move on to the next line
+					continue;
+				}
+
+				byte[] lineBytes = Encoding.UTF8.GetBytes(line);
+
+				for (int lineBytesRead = 0; lineBytesRead < lineBytes.Length;) {
+					if (messagePart.Length == 0) {
+						if (prefixBytes > 0) {
+							messagePartBytes += prefixBytes;
+							messagePart.Append(steamMessagePrefix);
+						}
+					} else {
+						messagePartBytes += NewlineWeight;
+						messagePart.AppendLine();
+					}
+
+					int bytesToTake = Math.Min(maxMessageBytes - messagePartBytes, lineBytes.Length - lineBytesRead);
+
+					// We can never have more characters than bytes used, so this covers the worst case of 1-byte characters exclusively
+					char[] lineChunk = charPool.Rent(bytesToTake);
+
+					try {
+						// We have to reset the decoder prior to using it, as we must discard any amount of bytes read from previous incomplete character
+						decoder.Reset();
+
+						int charsUsed = decoder.GetChars(lineBytes, lineBytesRead, bytesToTake, lineChunk, 0, false);
+
+						switch (charsUsed) {
+							case <= 0:
+								throw new InvalidOperationException(nameof(charsUsed));
+							case >= 2 when (lineChunk[charsUsed - 1] == '\\') && (lineChunk[charsUsed - 2] != '\\'):
+								// If our message is of max length and ends with a single '\' then we can't split it here, because it escapes the next character
+								// Instead, we'll cut this message one char short and include the rest in the next iteration
+								charsUsed--;
+
+								break;
+						}
+
+						int bytesUsed = Encoding.UTF8.GetByteCount(lineChunk, 0, charsUsed);
+
+						if (lineBytesRead > 0) {
+							messagePartBytes += ContinuationCharacterBytes;
+							messagePart.Append(ContinuationCharacter);
+						}
+
+						lineBytesRead += bytesUsed;
+
+						messagePartBytes += bytesUsed;
+						messagePart.Append(lineChunk, 0, charsUsed);
+					} finally {
+						charPool.Return(lineChunk);
+					}
+
+					if (lineBytesRead < lineBytes.Length) {
+						messagePartBytes += ContinuationCharacterBytes;
+						messagePart.Append(ContinuationCharacter);
+					}
+
+					// Check if we still have room for one more line
+					if (messagePartBytes + NewlineWeight + ReservedEscapeMessageBytes <= maxMessageBytes) {
+						continue;
+					}
+
+					yield return messagePart.ToString();
+
+					messagePartBytes = 0;
+					messagePart.Clear();
+				}
+			}
+
+			if (messagePart.Length == 0) {
```

Comment:
I just saw, that the previous one wasn't the last one. My bad.
```suggestion
			if (messagePart.Length == steamMessagePrefix.Length) {
```

Reply:
Then actually test your suggestions because they blow up only on 2 independent unit tests. You're not helping, and 6 standalone suggestions for a single thing I'm not going to accept either, send a diff or don't suggest code edits at all.